engine/utils/collections/Iterators.h

```diff
@@ -150,7 +150,7 @@ class MIter
      * @brief The dereference operator
      * @return the de-referenced pointer value
      */
-    inline constexpr T& operator*()
+    inline constexpr T& operator*() const
     {
         return *(ptr->Data() + index);
     }
```

Comment:
Would this force the dereferenced value to const?

Reply:
This is solely for when you're iterating over a `const` iterator. This would only work if the iterator itself is a `const`. Notice that the return type is still a normal references here.